### PR TITLE
Re-implement using pip-tools

### PR DESCRIPTION
**Pull request by AdrianDAlessandro**
This PR reimplements the poetry template as a pip-tools based template to provide us the alternative.

### CODE REVIEW

**Review by alexdewar — APPROVED** [review]
LGTM!

Why did you update the config files for the check links stuff btw? Was it not working for you?

**Review by dalonsoa — APPROVED** [review]
Looks great!

**Comment by AdrianDAlessandro** [comment]
> Why did you update the config files for the check links stuff btw? Was it not working for you?

It was working, but there was needless repetition but having one activated on push in `ci.yml` and a separate one for scheduled runs. I removed the push-activated one to reduce clutter in `ci.yml` and added the ability for it to run on push in the scheduled one.

File: README.md [hunk 1]
```diff
@@ -1,45 +1,67 @@
...
+This is a minimal Python 3.10 application that uses [`pip-tools`] for packaging and dependency management. It also provides [`pre-commit`](https://pre-commit.com/) hooks (for [`isort`](https://pycqa.github.io/isort/), [`black`](https://black.readthedocs.io/en/stable/), [`flake8`](https://flake8.pycqa.org/en/latest/) and [`mypy`](https://mypy.readthedocs.io/en/stable/)) and automated tests using [`pytest`](https://pytest.org/) and [GitHub Actions](https://github.com/features/actions). Pre-commit hooks are automatically kept updated with a dedicated GitHub Action, this can be removed and replace with [pre-commit.ci](https://pre-commit.ci) if using an public repo. It was developed by the [Imperial College Research Computing Service](https://www.imperial.ac.uk/admin-services/ict/self-service/research-support/rcs/).
+
+[`pip-tools`] is chosen as a lightweight dependency manager that adheres to the [latest standards](https://peps.python.org/pep-0621/ using `pyproject.toml`).
```
Comment (cc-a) [inline README.md]: The link at the end doesn't look right.

File: README.md [hunk 2]
```diff
@@ -1,45 +1,67 @@
...
+   - `pyproject.toml` (also change the list of authors here)
    - `tests/test_myproject.py`
    - Rename `myproject` directory
-6. Set up the virtual environment:
-
+5. Create and activate a Virtual Environment:
    ```bash
-   poetry install
+   python -m venv .venv
+   source .venv/bin/activate # on Windows: `.venv\Scripts\Activate.ps1`
```
Comment (cc-a) [inline README.md]:
```suggestion
   source .venv/bin/activate # with Powershell on Windows: `.venv\Scripts\Activate.ps1`

```

**Review by cc-a — APPROVED** [review]
Awesome. Thanks for this @AdrianDAlessandro. Couple of very tiny comments but otherwise good to go.

File: README.md (hunk 1 above)
Comment (AdrianDAlessandro) [reply README.md]: Interesting that the combination of markdownlint and check-links action didn't pick up on that

**Pull request merged**